tests/testthat/test-tokens_select.R

```diff
@@ -114,16 +113,207 @@ test_that("fcm works on tokens containing padding", {
+    expect_equal(
+        as.list(tokens_select(toks_uni, list(c("a b", "c"), "g j"))),
+        list(d1 = character(), d2 = character())
+    )
+    expect_equal(
+        as.list(tokens_select(toks_uni, list(c("a b", "c"), "g j"), padding = TRUE)),
+        list(d1 = rep("", 7), d2 = rep("", 7))
+    )
+})
+
+
+test_that("tokens_select works as expected for bigrams selected on char, list of bigrams", {
+    expect_equal(
+        as.list(tokens_select(toks_bi, char_bi)),
+        list(d1 = "a b", d2 = "a b")
+    )
+    expect_equal(
+        as.list(tokens_select(toks_bi, list_bi)), 
+        list(d1 = "a b", d2 = "a b")
+    )
+    expect_equal(
+        tokens_select(toks_bi, list_bi), 
+        tokens_select(toks_bi, char_bi)
+    )
+    expect_equal(
+        as.list(tokens_select(toks_bi, list(c("a b", "b e"), "g j"))),
+        list(d1 = character(0), d2 = c("a b", "b e"))
+    )
 })
 
+test_that("tokens_select works correctly with collocations objects", {
+    expect_equal(
+        as.list(tokens_select(toks_uni, coll_bi)),
+        list(d1 = character(), d2 = character())
+    )
+    expect_equal(
+        as.list(tokens_remove(toks_uni, phrase(coll_bi))),
+        list(d1 = c("c", "d"), d2 = c("i", "j"))
+    )
+    expect_equal(
+        as.list(tokens_select(toks_uni, coll_tri)),
+        list(d1 = character(), d2 = character())
+    )
+    expect_equal(
+        as.list(tokens_select(toks_bi, coll_bi)),
+        list(d1 = c("a b", "e g", "g h"), d2 = c("a b", "e g", "g h"))
+    )
+    expect_silent(
+        tokens_select(toks_bi, coll_bi)
+    )
+    expect_equal(
+        as.list(tokens_select(toks_bi, coll_tri)),
+        list(d1 = character(), d2 = character())
+    )
+    expect_silent(
+        tokens_select(toks_bi, coll_tri)
+    )
+})
+
+
+test_that("tokens_select fails as expected with dfm objects", {
+    expect_error(tokens_select(toks_uni, dfm_uni))
+    expect_error(tokens_select(toks_uni, dfm_bi))
+    expect_error(tokens_select(toks_bi, dfm_uni))
+    expect_error(tokens_select(toks_bi, dfm_bi))
+})
+
+
+test_that("tokens_select on unigrams works as expected when padding = TRUE", {
+    expect_equal(
+        as.list(tokens_select(toks_uni, "c d e", padding = TRUE)),
+        list(d1 = rep("", 7), d2 = rep("", 7))
+    )
+    expect_equal(
+        as.list(tokens_select(toks_uni, list("c d e"), padding = TRUE)),
+        list(d1 = rep("", 7), d2 = rep("", 7))
+    )
+    expect_equal(
+        as.list(tokens_select(toks_uni, list(c("c", "d", "e")), padding = TRUE)),
+        list(d1 = c("", "", "c", "d", "e", "", ""), d2 = rep("", 7))
+    )
+    
+    expect_equal(
+        as.list(tokens_select(toks_uni, coll_bi, padding = TRUE)),
+        list(d1 = rep("", 7), d2 = rep("", 7))
+    )
+    expect_equal(
+        as.list(tokens_select(toks_uni, list_bi, padding = TRUE)),
+        list(d1 = rep("", 7), d2 = rep("", 7))
+    )
+})
+
+test_that("tokens_select on bigrams works as expected when padding = TRUE", {
```

Comment:
No, I think there should be 7 pads, since 
```r
> toks_uni
tokens from 2 documents.
d1 :
[1] "a" "b" "c" "d" "e" "g" "h"

d2 :
[1] "a" "b" "e" "g" "h" "i" "j"
```
and none match the elements in the list `list_bi`.